plagiarism/http.go

```diff
@@ -0,0 +1,112 @@
+package plagiarism
+
+import (
+    "net/http"
+    "context"
+    "time"
+    "encoding/json"
+    "os/signal"
+    "os"
+    "log"
+)
+
+// Response struct will be used to transform to JSON format
+type Response struct {
+    Status string
+    Results map[string]string
+}
+
+// Server init variable
+type Server struct {
+    SubmissionDir string
+    ResultDir string
+    ID uint32
+    Logger *log.Logger
+    Port string
+}
+
+
+// Handler GET and POST in server
+func (s *Server) Handler() http.Handler {
+    return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+
+        // gluons only supports GET and POST reques to / path
+        // Throw HTTP error when path other than / is requested
+        if r.URL.Path != "/" {
+            http.Error(w, http.StatusText(http.StatusNotFound), http.StatusNotFound)
+            return
+        }
+
+        // Decode http request into variable
+        guids, err := DecodeRequest(r.Body)
+        if err != nil {
+            http.Error(w, http.StatusText(http.StatusNotAcceptable), http.StatusNotAcceptable)
```

Comment:
This status should only be used if the `Accept` header was sent and the server could not handle anything that the client wanted: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/406

It's better to use [`StatusBadRequest`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400).